### PR TITLE
machines: handle pools which don't support CreateVol API in add disk dialog

### PR DESCRIPTION
**Pull request by KKoukiou**
Show them in the dropdown list but disable the pools for which the
CreateVol API is not supported.

Relevant to https://bugzilla.redhat.com/show_bug.cgi?id=1731849

### CODE REVIEW

File: pkg/machines/components/diskAdd.jsx [hunk 1]
```diff
@@ -265,14 +265,15 @@ class PerformanceOptions extends React.Component {
 
 const CreateNewDisk = ({ idPrefix, onValueChanged, dialogValues, vmStoragePools, provider, vm }) => {
     const storagePool = vmStoragePools.find(pool => pool.name == dialogValues.storagePoolName);
+    const poolTypesNoSupportVolumeCreation = ['iscsi', 'iscsi-direct'];
```
Comment (janotomko) [inline pkg/machines/components/diskAdd.jsx]: In addition to those, neither gluster nor mpath support volume creation.
Also, "poolTypesNotSupportingVolumeCreation" would sound more English.

**Comment by marusak** [comment]
Needs to rebase to master since #12367 changed tests names

**Comment by KKoukiou** [comment]
> Needs to rebase to master since #12367 changed tests names

It's clean from machines failures, I would skip rebasing unless there are some review comments.

File: pkg/lib/cockpit-components-select.jsx [hunk 2]
```diff
@@ -92,13 +92,15 @@ Select.propTypes = {
  * Expected properties:
  *  - data (required), will be passed to the select's onChange callback
  *  - disabled (optional): whether or not the entry is disabled.
+ *  - title (optional): acts as a tooltip to the option entry
```
Comment (martinpitt) [inline pkg/lib/cockpit-components-select.jsx]: This collides with (almost but not quite) the same change in PR #12393. But fine to land this PR first, and rebase the other one.

**Pull request merged**